modules/javafx.controls/src/main/java/javafx/scene/control/skin/TabPaneSkin.java

```diff
@@ -842,13 +847,13 @@ public TabHeaderArea() {
                     if (tabsFit()) {
                         setScrollOffset(0.0);
                     } else {
-                        if (isSelectingTab) {
+                        if (scrollOffsetDirty) {
                             ensureSelectedTabIsVisible();
+                            scrollOffsetDirty = false;
                         } else {
-                            validateScrollOffset();
                         }
+                        validateScrollOffset();
```

Comment:
This seems to be an unintended change. If so please revert.

Reply:
good eye :) Forgot to remove the empty else block.

Validate must be called always if the tabs are not fitting: needed if the last tab (in the list) is visible and we remove tabs from the end. Without, the now last tab is not kept glued to the trailing edge. There are two tests covering the corner case (testRemoveXXLast) which fail if we don't.

Question is if that should be mentioned in a code comment? Personally, I tend to just delete the empty block.

Reply:
Thanks, I confirmed that it is a needed change. I think, a comment will be useful here, as the problem that is solved by this change is not very vivid from source.

Reply:
added a comment to clarify what the validate is doing